### ISSUE TITLE
Native support for Torchmetrics MetricCollection

### ISSUE THREAD

**Issue by SkafteNicki** [Lightning-AI/lightning#13254]
## 🚀 Feature

Add native support in `self.log` for the `MetricCollection` object from Torchmetrics:
https://torchmetrics.readthedocs.io/en/latest/pages/overview.html#metriccollection

### Motivation

The `MetricCollection` object is used to chain together multiple metrics into a single object that can be update at once:
```python
import torchmetrics as tm
metrics = tm.MetricCollection([tm.Accuracy(), tm.Precision(), tm.Recall()])
metrics.update(preds, target)
```
however, when one want to log such an object in PL it is often done on the individual metric:
```python
self.log_dict({k:v for k,v in collection.items()})
```
However, this actually does not work with the `compute_group` feature that we [implemented in v0.8](https://devblog.pytorchlightning.ai/torchmetrics-v0-8-paper-faster-collection-and-more-metrics-4a040619eb1d) for faster computation of metrics that share the same computations.

This issue proposed that PL natively supports `MetricCollection` in the same way that `Metric` is support in `self.log`.

### Additional context

Already discussed on slack with @Borda and @justusschock 
______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

- [**Metrics**](https://github.com/PyTorchLightning/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/PyTorchLightning/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/PyTorchLightning/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/PyTorchLightning/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

cc @borda @carmocca @edward-io @ananthsub @rohitgr7 @kamil-kaczmarek @Raalsky @Blaizzy @akihironitta

**Comment by carmocca** [comment]
Setting this to "future" for now as it might not make the 1.7 release
